iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java

```diff
@@ -265,9 +287,12 @@ public void commitAlterTable(org.apache.hadoop.hive.metastore.api.Table hmsTable
       HiveTableUtil.importFiles(preAlterTableProperties.tableLocation, preAlterTableProperties.format,
           partitionSpecProxy, preAlterTableProperties.partitionKeys, catalogProperties, conf);
     } else {
-      Map<String, String> contextProperties = context.getProperties();
-      if (contextProperties.containsKey(ALTER_TABLE_OPERATION_TYPE) &&
-          allowedAlterTypes.contains(contextProperties.get(ALTER_TABLE_OPERATION_TYPE))) {
+      if (isMatchingAlterOp(AlterTableType.ADDCOLS, context) && updateSchema != null) {
```

Comment:
Wouldn't it be cleaner to just get the operation in the beginning and use that through the method. I find it confusing that we call `isMatchingAlterOp` all the time

Reply:
`updateSchema` can only be non-null if `isMatchingAlterOp(AlterTableType.ADDCOLS, context)` was true in preAlterTable, so maybe `if (updateSchema != null)` is enough here? That might make this part also work generically for other schema update operations, like drop and rename, since you just have to call `commit()` here regardless of the operation type.

Reply:
Ok this is now stored as a state of this hook.
I usually like to be more restrictive, so I'd rather leave the ADDCOL op type check (btw I'm not sure how valid this case is, but if there are no new columns, then we have a case for ADDCOL op type with null updateSchema ;) )